File: ruby/Interpreter/subtract_expression.rb
require_relative 'expression'

class SubtractExpression < Expression
  attr_accessor :leftExpression
  attr_accessor :rightExpression

  def initialize(left, right)
    @leftExpression = left
    @rightExpression = right
  end

  # interpret each number and subtract them
  def interpret
    return @leftExpression.interpret - @rightExpression.interpret
  end
end

File: java/Interpreter/src/SubtractExpression.java
public class SubtractExpression implements Expression
{
    Expression leftExpression;
    Expression rightExpression;

    public SubtractExpression(Expression left, Expression right)
    {
        leftExpression = left;
        rightExpression = right;
    }

    public int interpret()
    {
        return leftExpression.interpret() - rightExpression.interpret();
    }
}